### PR TITLE
feat(state): supports state sync using delta changes

### PR DESCRIPTION
**Pull request by sebastijankuzner**
## Summary

Export & import changes.
Sync changes from client to server. 

## Checklist

- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: packages/state/source/repository.ts [hunk 1]
```diff
@@ -111,6 +111,48 @@ export class Repository implements Contracts.State.Repository {
...
+			if (this.#setAttributes.has(name)) {
+				continue;
+			}
+
+			forget.push(name);
+		}
+
+		for (const name of this.#setAttributes) {
+			set[name] = this.getAttributeHolder(name).toJson();
+		}
```
Comment (oXtxNt9U) [inline packages/state/source/repository.ts:128]: for-loops can be unified

File: packages/state/source/repository.ts [hunk 2]
```diff
@@ -111,6 +111,48 @@ export class Repository implements Contracts.State.Repository {
 		}
 	}
 
+	public changesToJson(): Contracts.State.RepositoryChange {
+		const set = {};
+		const forget: string[] = [];
+
+		for (const name of this.#setAttributes) {
+			if (this.#setAttributes.has(name)) {
```
Comment (oXtxNt9U) [inline packages/state/source/repository.ts:119]: Is this check correct? Doesn't it always evaluate to `true` ?

**Review by oXtxNt9U — CHANGES_REQUESTED** [review]

File: packages/state/source/repository.ts (hunk 2 above)
Comment (sebastijankuzner) [reply packages/state/source/repository.ts:119]: No. Mainsail core can register additional attributes that are not required by transactions. In this case we can skip it. For example validatorMatrix, etc.

File: packages/state/source/repository.ts (hunk 1 above)
Comment (sebastijankuzner) [reply packages/state/source/repository.ts]: Actually first should loop trough forget attributes. Fixed.

**Pull request merged**